awx_collection/plugins/modules/tower_project_update.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_project_update
+author: "Sean Sullivan (@sean-m-sullivan)"
+short_description: Sync a Project in Ansible Tower
+description:
+    - Sync an Ansible Tower Project. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - The name of the project to update.
+      required: True
+      type: str
+      aliases:
+        - project
+    organization:
+      description:
+        - Organization the project exists in.
+        - Used to help lookup the object, cannot be modified using this module.
+        - If not provided, will lookup by name only, which does not work with duplicates.
+      type: str
+    wait:
+      description:
+        - Wait for the project to update.
+      default: True
+      type: bool
```

Comment:
If we are going to make wait optional do we need a second tower_project_wait (like tower_job_launch and tower_job_wait) so that you can launch a project update, do some stuff in the background and then bring the wait into the foreground to get the results? This is really the async function in Ansible but I think our modules and pattern was developer before async was an actual thing.

Reply:
Good question, I had more thought of it, that You can wait for the results, but a given Job template is blocked until the corresponding project is updated, so if you launch a template afterwards it is still effectively doing the wait. Leaving the option up to what the user decides.